qemu/tests/qemu_guest_agent.py

```diff
@@ -1627,6 +1627,70 @@ def check_vss_info(cmd_type, key, expect_value):
                 test.error("guest-fsfreeze-thaw cmd failed with:"
                            "('%s')" % str(detail))
 
+    @error_context.context_aware
+    def gagent_check_fsinfo(self, test, params, env):
+        """
+        Execute "guest-fsinfo" command to guest agent,check file system of
+        mountpoint,disk's name and serial number.
+
+        steps:
+        1) check file system type of every mount point.
+        2) check disk name.
+        3) check disk's serial number.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+
+        """
+        session = self._get_session(params, None)
+        self._open_session_list.append(session)
+
+        error_context.context("Check all file system info in a loop.", logging.info)
+        fs_info_qga = self.gagent.get_fsinfo()
+        for fs in fs_info_qga:
+            mount_pt = fs["mountpoint"]
+            if params["os_type"] == "windows":
+                mount_pt = mount_pt[:2]
+
+            error_context.context("Check file system type of '%s' mount point." %
+                                  mount_pt, logging.info)
+            fs_type_qga = fs["type"]
+            cmd_get_disk = params["cmd_get_disk"] % mount_pt
+            disk_info_guest = session.cmd(cmd_get_disk).strip().split()
+            fs_type_guest = disk_info_guest[1]
+            if fs_type_qga != fs_type_guest:
+                test.fail("File System doesn't match.\n"
+                          "from guest-agent is %s,from guest os is %s."
+                          % (fs_type_qga, fs_type_guest))
+            else:
+                logging.info("File system type is %s which is expected." % fs_type_qga)
+
+            error_context.context("Check disk name.", logging.info)
+            disk_name_qga = fs["name"]
+            disk_name_guest = disk_info_guest[0]
+            if params["os_type"] == "linux":
+                if not re.findall(r'^/\w*/\w*$', disk_name_guest):
+                    disk_name_guest = session.cmd("readlink %s" % disk_name_guest).strip()
+                disk_name_guest = disk_name_guest.split('/')[-1]
+            if disk_name_qga != disk_name_guest:
+                test.fail("Device name doesn't match.\n"
+                          "name from guest-agent is %s,name from guest os is %s."
+                          % (disk_name_qga, disk_name_guest))
+            else:
+                logging.info("Disk name is %s which is expected." % disk_name_qga)
+
+            error_context.context("Check serial number of some disk.", logging.info)
+            serial_qga = fs["disk"][0]["serial"]
+            serial_param = params["blk_extra_params"]
```

Comment:
This can be move out of the loop.

Reply:
I don't know what you mean..

Reply:
It is same for all the device, no need to put it in  `for fs in fs_info_qga` to get it for every `fs`

Reply:
only this test case need test "serial number" of disk. So make it here is ok.

Reply:
@xiagao  Let me be more clear,  we are only talk about case `check_fsinfo`, only related with function `gagent_check_fsinfo`.   I am talking about no need to get this param in this loop ` L1651       for fs in fs_info_qga:`, because it is same for all `fs`.

Reply:
oh,I get what you mean.You are right, move this parameter out of this loop.